lib/services/event_service.ex

```diff
@@ -14,27 +14,31 @@ defmodule Aprb.Service.EventService do
     Poison.decode!(event.value)
   end
 
-  def process_event(event, topic) do
-    Task.start_link(fn -> SummaryService.update_summary(topic, event) end)
-    case topic do
+  def process_event(event, topic_name) do
+    topic = Repo.get_by(Topic, name: topic_name)
+    summary_task = Task.async(fn -> SummaryService.update_summary(topic, event) end)
```

Comment:
Here we create a asnyc task for updating summary, `fn` is basically lambda functions in ruby. Here, most of the code below this for processing other topics don't show summary so they don't have to wait for this background call to finish. Subscription topic though will show summary and in order for that to include the updated summary we need to wait for that async task to finish and then calculate monthly summary.

http://blog.danielberkompas.com/2016/04/05/background-jobs-in-phoenix.html